priv/repo/migrations/20161220162331_trigger_user_referential_integrity.exs

```diff
@@ -0,0 +1,13 @@
+defmodule Cog.Repo.Migrations.TriggerUserReferentialIntegrity do
+  use Ecto.Migration
+
+  def change do
+    execute """
+    ALTER TABLE triggers
+    ADD CONSTRAINT triggers_as_user_fkey
+    FOREIGN KEY(as_user) REFERENCES users(username)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE
```

Comment:
We talked about that, but decided to keep the username for now since it was more compatible with the current schema, and using the ID would require a more complicated migration and additional code modifications. It's also as unique as the ID.